go/vt/vttablet/tabletserver/query_executor.go

```diff
@@ -716,10 +716,13 @@ func (qre *QueryExecutor) execSelect() (*sqltypes.Result, error) {
 				q.SetErr(err)
 			}
 		} else {
-			qre.logStats.QuerySources |= tabletenv.QuerySourceConsolidator
-			startTime := time.Now()
-			q.Wait()
-			qre.tsv.stats.WaitTimings.Record("Consolidations", startTime)
+			maxwait := qre.tsv.config.ConsolidatorQueryWaiterCap
+			if maxwait == 0 || qre.tsv.qe.consolidator.Count(sqlWithoutComments) <= maxwait {
```

Comment:
The count method will slow down the execution query path.
We should have an absolute count rather than per query count or we need a fast path.

Reply:
> ... We should have an absolute count rather than per query count or we need a fast path.

Thanks Harshit, this is a slightly different point, but I agree a absolute count to some extend make more sense. I will think about this.